Gems/ProteusRobot/gem.json

```diff
@@ -22,7 +22,10 @@
     "documentation_url": "https://www.o3de.org/docs/user-guide/interactivity/robotics/project-configuration/#ros-2-project-templates",
     "dependencies": [],
     "repo_uri": "https://raw.githubusercontent.com/o3de/o3de-extras/development",
-    "compatible_engines": [],
+     "compatible_engines": [
+        "o3de-sdk>=1.2.0",
+        "o3de>=1.2.0"
```

Comment:
This is new to me, thank you for letting me know! I certainly did not intend to include the release version. I think we should exclude it.

Reply:
The [current release version is GitHub is `1.2.0`](https://github.com/o3de/o3de/blob/main/engine.json#L7) and the release SDK is `23.05.0`,  If you don't want to include those versions you would need to use `o3de>1.2.0` but unfortunately, because of the bug with the release SDK engine name being `o3de` instead of `o3de-sdk` that will include the release SDK version and the `compatible_engines` field is set up to indicate compatibility and not incompatibility so using a rule like `"o3de!=23.05.0"` would mean "compatible with every version of `o3de` except `23.05.0` which is not what you want because it includes `1.2.0`.  I think there isn't a simple way to include all future versions of the engine except one with the current rules.  In the future, we could add a syntax for that using something like `o3de>1.2.0,o3de!=23.05.0` but that is not currently supported.

Reply:
It seems that for now, we need to accept that this rule won't solve the issue fully, but at least it does help to define required versions a bit. I will merge it now, we will revisit this when the new syntax is there.